main.py

```diff
@@ -306,34 +329,71 @@ def main(cl_arguments):
             # This logic looks strange. We think it works.
             best_epoch = best_epoch[task.name]
             layer_path = os.path.join(args.run_dir, "model_state_eval_best.th")
-            load_model_state(
-                model,
-                layer_path,
-                args.cuda,
-                skip_task_models=task_names_to_avoid_loading,
-                strict=strict)
+            if args.transfer_paradigm == "finetune":
+                # save this fine-tune model with a task specific name
+                finetune_path = os.path.join(args.run_dir, "model_state_%s_best.th" % task.name)
+                os.rename(layer_path, finetune_path)
+
+                # reload the original best model
+                pre_finetune_path = get_best_checkpoint_path(args.run_dir)
+                load_model_state(model, pre_finetune_path, args.cuda, skip_task_models=[], strict=strict)
+            else:
+                load_model_state(model, layer_path, args.cuda,
+                                 skip_task_models=task_names_to_avoid_loading,
+                                 strict=strict)
 
     if args.do_full_eval:
         # Evaluate #
         log.info("Evaluating...")
-        val_results, val_preds = evaluate.evaluate(model, target_tasks,
-                                                   args.batch_size,
-                                                   args.cuda, "val")
-
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
-        if 'val' in splits_to_write:
-            evaluate.write_preds(target_tasks, val_preds, args.run_dir, 'val',
-                                 strict_glue_format=args.write_strict_glue_format)
-        if 'test' in splits_to_write:
-            _, te_preds = evaluate.evaluate(model, target_tasks,
-                                            args.batch_size, args.cuda, "test")
-            evaluate.write_preds(tasks, te_preds, args.run_dir, 'test',
-                                 strict_glue_format=args.write_strict_glue_format)
-        run_name = args.get("run_name", os.path.basename(args.run_dir))
-
-        results_tsv = os.path.join(args.exp_dir, "results.tsv")
-        log.info("Writing results for split 'val' to %s", results_tsv)
-        evaluate.write_results(val_results, results_tsv, run_name=run_name)
+        if args.transfer_paradigm == "finetune":
+            for task in tasks:
+                # TODO(Alex): special case for MNLI and MNLI-diagnostic
```

Comment:
Oh yea, this will mean we don't evaluate on MNLI-diagnostic

Issue is that we're assuming there's a fine-tuned model for each task. Quick fix would be searching for the MNLI fine-tund model and evaluating

Reply:
Should be resolved in PR #513